fail to build from source regression on freestanding + libc

### Zig Version

0.15.0-dev.1380+e98aeeb73

### Steps to Reproduce and Observed Behavior

Simple single-file reproduction reduced from one of my affected projects:
```zig
// Put this in a build.zig and run `zig build`
const std = @import("std");
pub fn build(b: *std.Build) void {
    const stub_file = b.addWriteFile("empty.zig", "\n");
    const stub_path = stub_file.getDirectory().path(b, "empty.zig");

    const target = b.resolveTargetQuery(.{
        .cpu_arch = .thumb,
        .os_tag = .freestanding,
        .cpu_model = .{ .explicit = &std.Target.arm.cpu.arm946e_s },
        .abi = .eabi,
    });

    const compiler_rt = b.addObject(.{
        .name = "compiler_rt_hack",
        .root_module = b.createModule(.{
            .target = target,
            .optimize = .Debug,
            .root_source_file = stub_path,
            .single_threaded = true,
            .link_libc = true,
        }),
    });

    compiler_rt.bundle_compiler_rt = true;
    compiler_rt.link_data_sections = true;
    compiler_rt.link_function_sections = true;
    b.default_step.dependOn(&b.addInstallFile(compiler_rt.getEmittedBin(), "asdf").step);
}
```

When running `zig build -freference-trace`:

```sh
/home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/posix.zig:5338:55: error: type 'void' has no members
    switch (errno(lseek_sym(fd, @bitCast(offset), SEEK.CUR))) {
                                                  ~~~~^~~~
referenced by:
    seekBy: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/fs/File.zig:1220:40
    sendFile: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Writer.zig:2208:31
    init: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Writer.zig:2177:31
    defaultDiscard: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Reader.zig:203:37
    Io.Reader.VTable: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Reader.zig:69:65
    VTable: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Reader.zig:23:20
    Io.Reader: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Reader.zig:16:16
    Reader: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io.zig:82:28
    fs.File.Reader: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/fs/File.zig:1127:22
    Reader: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/fs/File.zig:1117:20
    Io.Writer.VTable: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Writer.zig:63:27
    VTable: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Writer.zig:18:20
    Io.Writer: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io/Writer.zig:12:16
    Writer: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/Io.zig:83:28
    Writer: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/debug.zig:15:22
    panicExtra__anon_3789: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/debug.zig:594:13
    startGreaterThanEnd: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/std/debug.zig:50:33
    __atomic_load: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/compiler_rt/atomics.zig:122:17
    comptime: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/compiler_rt/atomics.zig:539:18
    comptime: /home/galaxyshard/.local/share/zigdown/zig-0.15.0-dev.1380+e98aeeb73/files/lib/compiler_rt.zig:252:65
```

Note that compiler_rt isn't the issue; it just happens to trigger a safety check (startGreaterThanEnd) which pulls in the std.debug.FullPanic, which uses std.Io.Writer.

### Expected Behavior

Successful compilation, as with versions 0.14.1 and before.

certainly not a recent regression 😆 
```
~ $ cat a.zig
pub fn main() void {
    const S = struct {
        foo: u32 = @compileError(""),
    };
    const s: S = .{ .foo = 10 };
    _ = s;
}
~ $ Downloads/zig-linux-x86_64-0.10.1/zig run a.zig -fno-stage1
a.zig:3:20: error:
        foo: u32 = @compileError(""),
                   ^~~~~~~~~~~~~~~~~
~ [1] $ Downloads/zig-linux-x86_64-0.10.1/zig run a.zig -fstage1
~ $
```

--- a/lib/std/fs/File.zig
+++ b/lib/std/fs/File.zig
@@ -1214,6 +1214,10 @@ pub const Reader = struct {
                     return err;
                 }
             }
+            if (posix.Stat == void) {
+                r.size_err = error.Streaming;
+                return error.Streaming;
+            }
             if (stat(r.file)) |st| {
                 if (st.kind == .file) {
                     r.size = st.size;
@@ -1236,6 +1240,10 @@ pub const Reader = struct {
                 setPosAdjustingBuffer(r, @intCast(@as(i64, @intCast(r.pos)) + offset));
             },
             .streaming, .streaming_reading => {
+                if (posix.SEEK == void) {
+                    r.seek_err = error.Unseekable;
+                    return error.Unseekable;
+                }
                 const seek_err = r.seek_err orelse e: {
                     if (posix.lseek_CUR(r.file.handle, offset)) |_| {
                         setPosAdjustingBuffer(r, @intCast(@as(i64, @intCast(r.pos)) + offset));

